libsolidity/formal/Predicate.cpp

```diff
@@ -334,8 +337,25 @@ vector<optional<string>> Predicate::localVariableValues(vector<smtutil::Expressi
 	auto first = _args.end() - static_cast<int>(localVars.size());
 	vector<smtutil::Expression> outValues(first, _args.end());
 
-	auto outTypes = applyMap(localVars, [](auto _var) { return _var->type(); });
-	return formatExpressions(outValues, outTypes);
+	auto inScope = [this](auto _var) -> bool {
+		for (auto _scope: m_scopeStack)
+		{
+			auto scope = dynamic_cast<ScopeOpener const*>(_var->scope());
+			solAssert(scope, "");
```

Comment:
The definition of `scope` can be moved outside the for loop.
Actually, this whole lambda could be replace by `std::find` for `_var->scope` and `m_scopeStack`. Not sure if there is any util functions here that would avoid the iterators in this case.

Reply:
Fixed